packages/cardpay-reward-api/cardpay_reward_api/indexer.py

```diff
@@ -84,8 +86,22 @@ def add_root_and_proofs(self, db: Session, root, payment_list):
                 validFrom=payment["validFrom"],
                 validTo=payment["validTo"],
             )
-            proofs.append(i)
-        db.bulk_save_objects(proofs)
+
+    def get_or_create(self, session, model, **kwargs):
+        """
+        Add db object if it doesn't exists
+        Ignore if it exists
+        """
+        kwargs.pop(
+            "proofArray", None
+        )  # proofArray is a list and cannot be used as a filter argument
+        instance = session.query(model).filter_by(**kwargs).first()
+        if instance:
+            return instance
+        else:
+            instance = model(**kwargs)
+            session.add(instance)
+            return instance
```

Comment:
This will be doing 2 queries per row now, is that right? Can we bulk upsert instead?